airbyte-webapp/src/components/connectorBuilder/Builder/KeyValueListField.tsx

```diff
@@ -60,29 +61,53 @@ interface KeyValueListFieldProps {
 export const KeyValueListField: React.FC<KeyValueListFieldProps> = ({ path, label, tooltip }) => {
   const [{ value: keyValueList }, , { setValue: setKeyValueList }] = useField<Array<[string, string]>>(path);
```

Comment:
Can't get rid of the "useField" here, but we can memoize everything happening _within_ the component if the key value list didn't actually change